src/services/EmailNotifications.php

```diff
@@ -87,18 +87,16 @@ protected function setLang(string $lang = 'en_GB'): void
 
     protected function getNotificationsToSend(): array
     {
-        // for step deadline only select notifications where deadline is in the next 30 min
+        // for step deadline only select notifications where deadline is in the next hour
```

Comment:
but then if you set the deadline in 1 hour, which is the first dropdown menu choice, you get an immediate notification, which seems weird. Why change from 30 minutes?

Reply:
Maybe it should also be an editable option on a per-deadline basis with a 30-minute default.

Some deadlines might only need a reminder relatively close to it, but others could benefit from a longer lead time, allowing users to adjust the reminder based on the importance or complexity of the task.

Reply:
An optional time of notif would be for later, along with a UI rebuild of this feature. Right now we want to remove the bug that the notif appears immediatly, so it must stay at 30 minutes.